feat: updated miscellaneous documentation (#51)
godot-escoria/escoria-docs: esc_reference.template.rst

``false`` and the item will not be checked for interactions. Its mouse
events won't be connected, either.

.. _global_flags:

Global flags
------------

Global flags (also known as game variables) define the state of the game and
can be true/false, a number, or a string. All commands and groups can be
conditionally set based on the value of a global flag.

Global flags are, as the name implies, global, and live past the room where
they are created/set. This means that a value set early in your game is still

> able to be queried many rooms later.

Reviewer: ```suggestion
Global flags are, as the name implies, global, and continue to survive after
leaving the room where they are created/set. This means that a value set
early in your game is still able to be queried many rooms later.
```

Global flags can be created anywhere in an Escoria script as needed using the

> set_global command. They don't need to be declared in advance.

Reviewer: ```suggestion
`set_global` command. They don't need to be declared in advance.
```

.. code-block::

   set_global <global name> <global value>

   e.g. set_global number_of_keys_found 3

By default a flag will return false if you haven't assigned it. This feature
allows for code like the below to configure a room the first time it's visited,

> but not to be rerun if you return.

Reviewer: ```suggestion
By default a flag will return false if you haven't declared or defined it. This
feature allows for code like the below--that configures a room--to be
executed the first time the room is visited, but won't be run again should
the player return to the room.
```

All ESC scripts are divided into a series of events which in turn run
commands or dialogs.

To use an event in your script, specify the name of the event preceded by a
colon. All commands following the event identifier are considered part of that
event until another event is defined in the same script file.

.. code-block::

   :push
   say player "I pushed the door."
   say player "It didn't do anything."

   # A new ("pull") event starts here
   :pull
   say player "The door is now open"

> In-built events

Reviewer: ```suggestion
Built-in events
```

~~~~~~~~~~~~~~~

Some events are hard-coded into Escoria. The ones that are
"internal engine use only" are
* print
* load
* room_selector
* transition_in
* transition_out

Events that are considered "for game developer use" are
* init : Run first as part of your primary Escoria game script. This is where
you would place the commands for a company logo cutscene.

> * exit_scene : Will be called when "Is Exit" is enabled on an ESCItem and the

Reviewer: ```suggestion
* exit_scene : Will be called when "Is Exit" is enabled on an `ESCItem` and the
```

player "uses" that item. You might play a closing door sound here for
example.
* newgame : This is what is called when "Start Game" is chosen from your menu.

> The main use would be to have a "change_scene" command here to load your

Reviewer: ```suggestion
The main use would be to have a `change_scene` command here to load your
```

first game room.
* setup : This runs first as part of loading a room. Anything coded here will
happen before the room is visible (i.e. before the "transition in").
.. _ready-label:
* ready : These are commands that will run when a room loads, after it becomes
visible (i.e. once ":setup" completes and after the "transition in").

Plugin Events
~~~~~~~~~~~~~

> Any plugins you use may define your their own events that you can script

Reviewer: ```suggestion
Any plugins you use may define their own events that you can script
```

actions for. The sample user interfaces, for example, include events for

> "look" and "use". If you are using the 9-verb interface and click the "look"

Reviewer: ```suggestion
`look` and `use`. If you are using the 9-verb interface and click the `look`
```

> button followed by an object, the any code under the ":look" event in that

Reviewer: ```suggestion
button followed by an object, then any code inside the `:look` event in that
```

> objects' script will be run.

Reviewer: ```suggestion
object's script will be run.
```

User-created events
~~~~~~~~~~~~~~~~~~~

You can create an event with any name you like (though avoiding terms Escoria

> uses is suggested to avoid bugs and/or confusion.)

Reviewer: ```suggestion
You can create an event with any name you like (though avoiding event names
Escoria already uses, e.g. `init`, is suggested to avoid bugs and/or confusion.)
```

> Most of the time you'll define events as part of creating your UI (e.g you

Reviewer: ```suggestion
Most of the time you'll define events as part of creating your UI (e.g. you
```
